src/components/Tree/Node.tsx

```diff
@@ -0,0 +1,91 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { IconProps, IconSize } from "@components/Icon/Icon";
+import { ReactComponent as IconCaretDown } from "@components/Icon/Svg/CaretDown.svg";
+import { ReactComponent as IconCaretRight } from "@components/Icon/Svg/CaretRight.svg";
+import { ReactElement, useEffect, useState } from "react";
+import css from "./Tree.module.css";
+
+export interface TreeNode {
+    id: string;
+    name: string;
+    icon?: ReactElement<IconProps>;
+    label?: string;
+    value?: string;
+    nodes?: TreeNode[];
+}
+
+interface NodeProps {
+    node: TreeNode;
+    strong?: boolean;
+    activeNodeId?: string;
+    onClick: (id: string) => void;
+}
+
+const getNodeClassNames = (active: boolean, strong: boolean, hasValue: boolean) => {
+    const classNames = [css.nodeLink];
+    if (active) {
+        classNames.push(css.nodeLinkSelected);
+    }
+    if (hasValue) {
+        classNames.push(css.selectableNodeLink);
+    }
+    if (strong) {
+        classNames.push(css.strong);
+    }
+
+    return classNames;
+};
+
+export default function Node({
+    node: { id, value, name, label, icon, nodes },
+    strong = false,
+    activeNodeId,
+    onClick,
+}: NodeProps): ReactElement<NodeProps> {
+    const [showNodes, setShowNodes] = useState(false);
+    const [active, setActive] = useState(false);
+
+    useEffect(() => setActive(id === activeNodeId), [activeNodeId]);
+
+    return (
+        <li data-test-id="node">
+            <a
+                className={getNodeClassNames(active, strong, !!value).join(" ")}
+                onClick={() => {
+                    if (value) {
+                        setActive(true);
+                        onClick && onClick(id);
```

Comment:
```suggestion
     return (
        <li data-test-id="node">
            <a
                className={getNodeClassNames(id === activeNodeId, strong, !!value).join(" ")}
                onClick={() => {
                    if (value) {
                        onClick && onClick(id);
```
I haven't tried but can't we bypass the `useEffect`?

Reply:
since it's used by the state, it's needed there

Reply:
@triggertoo I think @SamuelAlev is right since you pass in the `activeNodeId` and keep track of it outside within `Tree` you do not need to track it here with `useState` since if a new `activeNodeId` is passed in from outside the the `TreeNode` gets re-rendered.

Reply:
ah, now i see it 👍  adaf607